engine/verification/verifier/engine.go

```diff
@@ -195,22 +195,28 @@ func (e *Engine) verify(ctx context.Context, originID flow.Identifier,
 	}
 
 	// if any fault found with the chunk
+	sendApproval := true
 	if chFault != nil {
 		switch chFault.(type) {
 		case *chmodels.CFMissingRegisterTouch:
 			e.log.Error().Msg(chFault.String())
+			// still create approvals for this case
 		case *chmodels.CFNonMatchingFinalState:
 			// TODO raise challenge
 			e.log.Warn().Msg(chFault.String())
+			sendApproval = false
```

Comment:
can we `return nil` here without the `sendApproval` variable

Reply:
yeah I fixed it.